src/middlewared/middlewared/api/v25_04_0/nfs.py

```diff
@@ -0,0 +1,194 @@
+    userd_manage_gids: bool
+    """ Enable to allow server to manage gids """
+    keytab_has_nfs_spn: bool
+    """ Report status of NFS Principal Name in keytab (status only)"""
+    managed_nfsd: bool
+    """ Report status of 'servers' field.
+    If True the number of nfsd are managed by the server. (status only)"""
+    rdma: bool
+    """ Enable or disable NFS over RDMA.  Requires RDMA capable NIC """
+
+    @field_validator('bindip')
+    @classmethod
+    def check_bind_ip(cls, field_value: list):
+        """ Custom validator for IP addresses to avoid JSON serialization errors """
+        all(isinstance(v, IPvAnyAddress) for v in field_value)
+        return field_value
+
+
+@single_argument_args('nfs_update')
+class NfsUpdateArgs(NfsEntry, metaclass=ForUpdateMetaclass):
+    id: Excluded = excluded_field()
+    managed_nfsd: Excluded = excluded_field()
+    v4_krb_enabled: Excluded = excluded_field()
+    keytab_has_nfs_spn: Excluded = excluded_field()
+
+
+class NfsUpdateResult(BaseModel):
+    result: NfsEntry
+
+
+class NfsBindipChoicesArgs(BaseModel):
+    pass
+
+
+@single_argument_result
+class NfsBindipChoicesResult(BaseModel):
+    """ Return a dictionary of IP addresses """
+    model_config = ConfigDict(extra='allow')
+
+
+class NfsShareEntry(BaseModel):
+    id: int
+    path: NonEmptyString
+    """ Local path to be exported. """
+    aliases: list[NonEmptyString] = []
+    """ IGNORED for now """
+    comment: str = ""
+    """ User comment associated with share """
+    networks: list[NonEmptyString] = []
+    """ List of authorized networks that are allowed to access the share having format
+        "network/mask" CIDR notation. If empty, all networks are allowed. """
+    hosts: list[NonEmptyString] = []
+    """ list of IP's/hostnames which are allowed to access the share.
+        If empty, all IP's/hostnames are allowed. """
+    ro: bool = False
+    """ Export the share as read only. """
+    maproot_user: str | None = None
+    """ Map root user client to a specified user. """
+    maproot_group: str | None = None
+    """ Map root group client to a specified group. """
+    mapall_user: str | None = None
+    """ Map all client users to a specified user. """
+    mapall_group: str | None = None
+    """ Map all client groups to a specified group. """
+    security: list[Literal["SYS", "KRB5", "KRB5I", "KRB5P"]] = []
+    """ Specify the security schema """
+    enabled: bool = True
+    """ Enable or disable the share """
+    locked: bool | None
+
+    @field_validator('networks', 'hosts')
+    @classmethod
+    def check_unique(cls, field_value: list):
+        """ Custom validator to confirm unique entries """
+        s = set()
+        not_unique = []
+        for i, v in enumerate(field_value):
+            if v in s:
+                not_unique.append(v)
+            s.add(v)
+            if not_unique:
+                raise ValueError(f"Entries must be unique, the following are not: {not_unique}")
```

Comment:
You go to the trouble of having a `not_unique` **list**, and then error out on the first entry discovered.  Should the `if` block be de-indented?